climada/hazard/trop_cyclone.py

```diff
@@ -740,50 +743,31 @@ def _compute_windfields_sparse(
 
     # convert track variables to SI units
     si_track = tctrack_to_si(track, metric=metric)
-    t_lat, t_lon = si_track["lat"].values, si_track["lon"].values
-
-    # normalize longitudinal coordinates of centroids
-    u_coord.lon_normalize(coastal_centr[:, 1], center=si_track.attrs["mid_lon"])
-
-    # Restrict to the bounding box of the whole track first (this can already reduce the number of
-    # centroids that are considered by a factor larger than 30).
-    max_dist_eye_lat = max_dist_eye_km / u_const.ONE_LAT_KM
-    max_dist_eye_lon = max_dist_eye_km / (
-        u_const.ONE_LAT_KM * np.cos(np.radians(np.abs(coastal_centr[:, 0]) + max_dist_eye_lat))
-    )
-    coastal_idx = coastal_idx[
-        (t_lat.min() - coastal_centr[:, 0] <= max_dist_eye_lat)
-        & (coastal_centr[:, 0] - t_lat.max() <= max_dist_eye_lat)
-        & (t_lon.min() - coastal_centr[:, 1] <= max_dist_eye_lon)
-        & (coastal_centr[:, 1] - t_lon.max() <= max_dist_eye_lon)
-    ]
-    coastal_centr = centroids.coord[coastal_idx]
```

Comment:
The lines above are now moved to the `get_close_centroids` helper function. This has the advantage that this functionality does not need to be re-implemented in `TCRain` (in CLIMADA petals) and we can cover this part of the functionality with our unit tests more easily.